plugins/module_utils/network/eos/config/ospfv2/ospfv2.py

```diff
@@ -356,10 +360,16 @@ def add_commands(self, want, have):
                     commands.append(c)
             if ospf_params.get("auto_cost"):
                 commands.append(
-                    "auto-cose reference-bandwidth " + ospf_params["auto_cost"]
+                    "auto-cost reference-bandwidth " + ospf_params["auto_cost"]
                 )
             if ospf_params.get("bfd"):
-                commands.append("bfd default")
+                os_version = get_capabilities(self._module)["device_info"][
```

Comment:
I think if we go through the module_utils route, the Connection object will always be created and `rendered` will no longer work offline. At this point in code, the self._connection object should be set (expect for `rendered` and `parsed`), so we can possibly do something like:
```
if self._connection:
    os_version = self._connection.get_device_info()["network_os_version"]
```
For `rendered`, however, we can compute command strings based on what's valid for the standard EOS version and mention that in the doc.